spec/mem/tree_borrows/state_machine.md

```diff
@@ -44,49 +49,70 @@ impl Permission {
     fn child_read(self) -> Result<Permission> {
         ret(
             match self {
-                Permission::Reserved => Permission::Reserved,
-                Permission::Active => Permission::Active,
-                Permission::Frozen => Permission::Frozen,
                 Permission::Disabled => throw_ub!("Tree Borrows: reading from the child of a pointer with Disabled permission"),
+                _ => self,
             }
         )
     }
 
-    fn child_write(self) -> Result<Permission> {
+    fn child_write(self, protected: Protected) -> Result<Permission> {
         match self {
-            Permission::Reserved => ret(Permission::Active),
-            Permission::Active => ret(Permission::Active),
+            Permission::Reserved { conflicted: true } if protected.yes() =>
+                throw_ub!("Tree Borrows: writing to the child of a protected pointer with Conflicted Reserved permission"),
             Permission::Frozen => throw_ub!("Tree Borrows: writing to the child of a pointer with Frozen permission"),
             Permission::Disabled => throw_ub!("Tree Borrows: writing to the child of a pointer with Disabled permission"),
+            _ => ret(Permission::Active),
         }
     }
 
-    fn foreign_read(self) -> Result<Permission> {
-        ret(
+    fn foreign_read(self, protected: Protected, accessed: Accessed) -> Result<Permission> {
```

Comment:
In Miri, all transitions are given is a `protected: bool`. I think we should do the same here, let's have the smallest possible amount of degrees of freedom.

Reply:
Same reason as the above reply.